js/main.js

```diff
@@ -13,11 +13,45 @@ $avatarUrl.addEventListener('input', function (event) {
 
 $formSelect.addEventListener('submit', function (event) {
   event.preventDefault();
-  data.profile.username = $userNameInfo.value;
-  data.profile.avatarUrl = $avatarUrl.value;
-  data.profile.fullName = $userFullNameInfo.value;
-  data.profile.location = $userLocationInfo.value;
-  data.profile.bio = $userBioInfo.value;
+  if (!$userNameInfo.value.trim()) { // prevents user from having username with spaces
```

Comment:
This is a great idea, Phillip!! Rather than having a succession of conditionals doing the same work here though, this would be a great opportunity that will take in a value, and return `true` or `false`, depending on if it is empty after being trimmed. Maybe something like `isEmptyFormField`, where you return `true` if it is empty (since the functionality should match the name), and `false` if it is not. Then, you can just call that function here in this function for each value you need to check!